### PR TITLE
Support $LATEST replace for BQ Table type

### LINKED ISSUE #3363: Add method to resolve latest `Table` partition
Similar to #3357

### PR DESCRIPTION
**Pull request by liufuyang**
closes #3363

### CODE REVIEW

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala [hunk 1]
```diff
@@ -68,14 +68,23 @@ sealed trait Table extends Source {
   def spec: String
 
   def ref: GTableReference
+
+  def latest(bg: BigQuery): Table
```
Comment (regadas) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:72]: can we add `def latest()` as well?

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala [hunk 2]
```diff
@@ -68,14 +68,23 @@ sealed trait Table extends Source {
...
     override lazy val spec: String = BigQueryHelpers.toTableSpec(ref)
+    def latest(bq: BigQuery): Ref =
+      Ref(Spec(spec).latest(bq).ref)
+
   }
   final case class Spec(spec: String) extends Table {
     override val ref: GTableReference = BigQueryHelpers.parseTableSpec(spec)
+    def latest(bq: BigQuery): Spec = {
+      val specWithLatestReplaced = BigQueryPartitionUtil.latestTable(bq, spec)
+      Spec(specWithLatestReplaced)
```
Comment (regadas) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:86]:
```suggestion
      Spec(BigQueryPartitionUtil.latestTable(bq, spec))
```

Comment (regadas) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:86]: it's ok to be one liner

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala (hunk 1 above)
Comment (regadas) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:72]: Also some scaladoc as we did for `Query`?

Comment (liufuyang) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:72]: Indeed, sorry I forgot that one :)

Do you know where I can see some examples how how to use this `Table` type or what is the most common way to use it? Then I can perhaps write a little scala doc here about it as I saw that
https://github.com/spotify/scio/issues/3208

Comment (regadas) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:72]: You can have a look at `scio-examples` as we have some things there like https://github.com/spotify/scio/blob/4ca965e0c1d82f8aafb2628eaa92edb5b5f64337/scio-examples/src/main/scala/com/spotify/scio/examples/cookbook/BigQueryTornadoes.scala#L34

The migration guide also gives you an overview of where it's used https://spotify.github.io/scio/migrations/v0.9.0-Migration-Guide.html#bigquery.

Although for this PR is enough just to add doc around the latest.

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala [hunk 3]
```diff
@@ -64,18 +64,56 @@ final case class Query(underlying: String) extends Source {
   def latest(): Query = latest(BigQuery.defaultInstance())
 }
 
+/**
+ * A wrapper type [[Table]] which wraps a table spec String and a table [[GTableReference]]
```
Comment (regadas) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:68]:
```suggestion
 * [[Table]] abstracts the multiple ways of referencing Bigquery tables. Tables can be referenced by a table spec `String` or by a table reference [[GTableReference]]. 
```

Comment (regadas) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala:68]: @liufuyang I think this suits it better.

Comment (liufuyang) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryTypes.scala]: Indeed. I had the concept but not exactly sure how to phrase it. Thank you very much for the suggestion 👍

**Comment by liufuyang** [comment]
Looks like there is more code unrelated to my PR having format issues. 

Isn't there some command one can run to reformat everything on master branch and push in to solve this or?

**Review by regadas — APPROVED** [review]
Just missing a rebase an we are good to go!

**Pull request merged**